frontend/src/types/index.ts

```diff
@@ -0,0 +1,73 @@
+import type { Day } from "@/services/usos/types";
+
+export interface CreatePlanResponseType {
+  success: boolean;
+  message: string;
+  schedule: {
+    name: string;
+    userId: number;
+    createdAt: string;
+    updatedAt: string;
+    id: number;
+  };
+}
+
+export interface PlanResponseType {
+  name: string;
+  userId: number;
+  id: number;
+  createdAt: string;
+  updatedAt: string;
+  courses: Array<{
+    id: string;
+    name: string;
+    department: string;
+    lecturer: string;
+    type: string;
+    ects: number;
+    semester: number;
+    groups: Array<{
+      id: number;
+      name: string;
+      day: string;
+      time: string;
+      room: string;
+    }>;
+  }>;
+  registrations: Array<{
+    id: string;
+    name: string;
+  }>;
+}
+
+export interface DeletePlanResponseType {
+  success: boolean;
+  message: string;
+}
+
+export type CourseType = Array<{
```

Comment:
Pewnie przekopiowałeś to, ale nie możemy zrobić wszędzie interfejsów, żeby spójnie było?

Reply:
nie da się takiego typu zrobić interfacem niestety